cluster-operator/src/main/java/io/strimzi/operator/cluster/tracing/TracingUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.tracing;
+
+import io.strimzi.api.kafka.model.tracing.OpenTelemetryTracing;
+import io.strimzi.api.kafka.model.tracing.JaegerTracing;
+import io.strimzi.api.kafka.model.tracing.Tracing;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Handle any tracing abstraction
+ */
+public class TracingUtils {
+
+    private static final Map<Class<? extends Tracing>, TracingHandle> HANDLE_MAP;
+
+    static {
+        HANDLE_MAP = new HashMap<>();
+        HANDLE_MAP.put(JaegerTracing.class, new OpenTracingHandle());
+        HANDLE_MAP.put(OpenTelemetryTracing.class, new OTelTracingHandle());
+    }
+
+    private static TracingHandle getTracingHandle(Tracing tracing) {
+        Objects.requireNonNull(
+            tracing,
+            "Tracing configuration instance must not be null - missing null check in CR handling?"
+        );
+        Class<? extends Tracing> clazz = tracing.getClass();
+        TracingHandle handle = HANDLE_MAP.get(clazz);
+        if (handle == null) {
+            throw new IllegalArgumentException("No matching TracingHandle for type: " + clazz.getName());
+        }
+        return handle;
+    }
+
+    public static String consumerInterceptor(Tracing tracing) {
+        return getTracingHandle(tracing).consumerInterceptor();
+    }
+
+    public static String producerInterceptor(Tracing tracing) {
+        return getTracingHandle(tracing).producerInterceptor();
+    }
+
+    interface TracingHandle {
+        String consumerInterceptor();
+        String producerInterceptor();
```

Comment:
Maybe these should be `consumerInterceptorClassName()` and `producerInterceptorClassName()`?

Reply:
Those methods are long gone :-)

It's now only this one

```
    void addTracingPropsToProducerConfig(Properties props);
```

Reply:
Ah, sorry ... this is Op, not Bridge ... darn  ... let me fix it then.

Reply:
@tombentley and I guess we also probably need to check if user already set this interceptors.classes option, and just add to it not simply override it ?